src/lib/Hydra/Helper/AddBuilds.pm

```diff
@@ -105,10 +105,14 @@ sub fetchInputBuild {
     my $relName = ($prevBuild->releasename or $prevBuild->nixname);
     my $version = $2 if $relName =~ /^($pkgNameRE)-($versionRE)$/;
 
+    my $mainOutput = getMainOutput($prevBuild);
+
     return
-        { storePath => getMainOutput($prevBuild)->path
+        { storePath => $mainOutput->path
         , id => $prevBuild->id
         , version => $version
+        , outputName => $mainOutput->name
+        , drvPath => $prevBuild->drvPath
```

Comment:
Oh, if that's the case this is less useful. Previously I've been adding a build that generates a nix expression that imports the drv (taken as an input source via `unsafeDiscardOutputDependency`) as part of the jobset definition, which doesn't have this problem, but this would have been nicer. Any objection to keeping the drv as a root as long as the build is?